src/main/java/com/KacFlor/ShopSpring/service/ProductService.java

```diff
@@ -161,7 +164,7 @@ public boolean removePromotion(Integer PTid, Integer PNid){
             return true;
         }
         else{
-            throw new ExceptionsConfig.ResourceNotFoundException("Payment not found");
+            throw new ExceptionsConfig.ResourceNotFoundException("Resource not found");
```

Comment:
Lets divide it so we there is proper message, when optionalProduct is not present -> Product not found, when optionalPromotion is not present -> Promotion not found, do it as as early return, https://medium.com/swlh/return-early-pattern-3d18a41bba8, apply everywhere

Reply:
done